internal/lock/package.go

```diff
@@ -22,13 +26,30 @@ type Package struct {
 }
 
 type SystemInfo struct {
-	// StorePath is the input-addressed path for the nix package in /nix/store
-	// It is the cache key in the Binary Cache Store (cache.nixos.org)
-	// It is of the form /nix/store/<hash>-<name>-<version>
-	// <name> may be different from the canonicalName so we store the full store path.
+	Outputs []*Output `json:"outputs,omitempty"`
```

Comment:
Why pointers? If you make them structs you don't need the `Equals` function below you can just compare

`a == b`